### PR TITLE
number 7

### PR DESCRIPTION
**Pull request by gatorsgonnagait**

### CODE REVIEW

File: 8_Largest_product.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+def largest_products_series(limit, series):
+    # converts large number to a series of digits
+    series_digits = [int(d) for d in str(series)]
+    max_product = 1
+    for index, digit in enumerate(series_digits):
+
+        cut_series = series_digits[index:index+limit]
```
Comment (chenguo) [inline 8_Largest_product.py:7]: Don't you run off the edge of the array here? If index is `len(series) - 1` you don't get an exception here?

Comment (gatorsgonnagait) [reply 8_Largest_product.py:7]: hmm yea i dont know why it ran

Comment (gatorsgonnagait) [reply 8_Largest_product.py:7]: I think python slice just handles it by itself

Comment (chen-factual) [reply 8_Largest_product.py:7]: Yeah looks like it:

```python
>>> a = [0, 1, 2, 3, 4]
>>> a[3:5]
[3, 4]
>>> a[4:10]
[4]
>>> a[19]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```

it's not good to depend on implicit behavior like this. Generally speaking you should be more rigorous about understanding and controlling your iteration range. What if Python4 changes this behavior? What if you need to write some loops in another language?

Comment (gatorsgonnagait) [reply 8_Largest_product.py:7]: Then it would've thrown an index out of bounds exception and I would have fixed it?

Comment (chenguo) [reply 8_Largest_product.py:7]: Imagine you took this approach with all your code. Python makes an update, and all of a sudden you have to update _all_ of your dozens of projects and thousands of lines of code.

It's always good practice to code defensively when reasonable.

**Pull request merged**